python/src/dataset.rs

```diff
@@ -466,6 +466,10 @@ impl Dataset {
         Ok(())
     }
 
+    fn count_deleted_rows(self_: PyRef<'_, Self>) -> usize {
+        self_.ds.count_deleted_rows()
```

Comment:
nit: these new methods could all take `&self`, instead of `PyRef<...>`.

Reply:
fixed!